apollo-federation/src/sources/connect/validation/selection.rs

```diff
@@ -129,13 +132,18 @@ impl<'schema> JSONSelectionVisitor for SelectionValidator<'schema> {
 
     fn enter_group(&mut self, field_name: &str) -> Result<(), Self::Error> {
         let parent = self.path.last().copied().unwrap_or(self.root);
```

Comment:
Won't this always have a component now, since it's being initialized with the starting field?